Feature/catholic holidays

zadca123 opened the pull request:
Hi,
I added some catholic holidays. Not finished yet, i think the resolution with all? option is bad, and i should use special variable for this.

Question:
For what purpose there are nil's in your append? Like in the below example. It works without nil too, even with just one list
```
(defvar polish-holidays-notable
  (append polish-holidays-national
          polish-holidays-other
          nil)
  "Notable holidays and commemoration dates in Poland.")
```

also feel free to give some advices

Commit:
* polish-holidays.el: adding polish catholic holidays

Commit:
* polish-holidays.el: option to set all holidays

przemarbor commented:
Today I also wondered about `nil`'s in my code and to be honest: I have no idea what they are meant to do :smile:. It's probably an artefact taken from the repositories I was inspired by during my work. For example [here](https://github.com/petrem/romanian-holidays/blob/main/romanian-holidays.el) they use `nil` in the same context (line 137).

When it comes to your commits - thank you very much for contribution!  (I didn't have patience to list all those catholic holidays...).

However, there exist two issues:
1) In your implemetation, catholic holidays that coincide with already defined notable/national holidays
   appear in the calendar twice (Święto Trzech Króli, Wielkanoc, etc.).   I think it'd be better if we removed repeated entries from catholic-holiday lists  and changed their names appropriately
   (`polish-holidays-catholic--fixed-holidays` -> `polish-holidays-nonnational-catholic--fixed-holidays`, and
   `polish-holidays-catholic--paschal-cycle` -> `polish-holidays-nonnational-catholic--paschal-cycle`).
   
2) I don't like the name `polish-holiday-get` as it suggests that this function is kind of complementary to `polish-holiday-set` (in a way like setter is complementary to getter in objective programming), while the main purpose of the function  is to select which calendar one wants to use...

Commit:
removing nil's

Commit:
polish-holidays-catholic -> polish-holidays-nonnational-catholic rename

Commit:
better way to handle using all holidays and changing variable names

Commit:
removing redundant holidays

Commit:
adding basic readme

zadca123 commented:
Things done:
1. removed nils
2. Changed variable names
3. implementing variable switch to choose all or just notable holidays
4. no more redundant holidays
5. Adding basic README

przemarbor commented:
Thanks for contribution!
I merged your changes into main. There were three more issues to resolve. I did them by myself to avoid bothering you.
-  AFAIK Pentecost is national holiday - correct me if I'm wrong...
-  I deleted README.org and moved its content the comment part of `polish-holiday.el`.
   This is done in accordance with [melpa's guide](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org) where you can read:
   `Avoid change logs and readmes
    Files like these would only end up in an obscure installation directory where a user would never know to look when 
    searching for information. Possible exceptions to this policy are .info files since these are automatically added to 
    Emacs’ info index and the presence of interactive commands that would display the files.`
- I also modified the example of the default configuration to a more secular one :wink:
